pkg/security/podsecuritypolicy/seccomp/mustrunas.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package seccomp
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+const (
+	allowAnyProfile = "*"
+)
+
+// withSeccompProfile implements the SeccompStrategy.
+type withSeccompProfile struct {
+	allowedProfiles []string
+}
+
+var _ SeccompStrategy = &withSeccompProfile{}
+
+// NewWithSeccompProfile creates a new must run as strategy or returns an error if it cannot
+// be created.
+func NewWithSeccompProfile(allowedProfiles []string) (SeccompStrategy, error) {
+	return &withSeccompProfile{allowedProfiles}, nil
+}
+
+// Generate creates the profile based on policy rules.
+func (s *withSeccompProfile) Generate(pod *api.Pod) (string, error) {
+	// return the first non-wildcard profile
+	for _, p := range s.allowedProfiles {
+		if p != allowAnyProfile {
+			return p, nil
+		}
+	}
+	// if we reached this point then either there are no allowed profiles (empty slice)
+	// or the only thing in the slice is the wildcard.  In either case just return empty.
+	return "", nil
+}
+
+// ValidatePod ensures that the specified values on the pod fall within the range
+// of the strategy.
+func (s *withSeccompProfile) ValidatePod(pod *api.Pod) field.ErrorList {
+	allErrs := field.ErrorList{}
+	podSpecFieldPath := field.NewPath("pod", "spec", "securityContext", "seccompProfile")
+
+	if len(s.allowedProfiles) == 0 &&
+		pod.Spec.SecurityContext != nil &&
+		pod.Spec.SecurityContext.SeccompProfile != "" {
+
+		allErrs = append(allErrs, field.Forbidden(podSpecFieldPath, "seccomp may not be set"))
+		return allErrs
+	}
+
+	if pod.Spec.SecurityContext != nil {
+		if !isProfileAllowed(pod.Spec.SecurityContext.SeccompProfile, s.allowedProfiles) {
+			allErrs = append(allErrs, field.NotSupported(podSpecFieldPath.Child("securityContext", "seccompProfile"),
```

Comment:
the field path is wrong here. Above you also dive down into securityContext and seccompProfile.